### PR TITLE
HMIS Data Quality Updates

### PR DESCRIPTION
**Pull request by eanders**

### CODE REVIEW

File: db/warehouse/migrate/20221006193112_add_summary_fields_to_dqt_report.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
+class AddSummaryFieldsToDqtReport < ActiveRecord::Migration[6.1]
+  def change
+    add_column :hmis_dqt_clients, :veteran_status, :integer
+    add_column :hmis_dqt_clients, :ssn, :integer
```
Comment (awisema) [inline db/warehouse/migrate/20221006193112_add_summary_fields_to_dqt_report.rb]: Should this be a string?

File: drivers/hmis_data_quality_tool/app/models/hmis_data_quality_tool/client.rb [hunk 2]
```diff
@@ -294,31 +314,100 @@ def self.sections
...
+          required_for: 'All',
+          denominator: ->(_item) { true },
+          limiter: ->(item) {
+            # SSN DQ is 99
+            return true if item.ssn_data_quality == 99 || item.ssn_data_quality.blank?
+            # SSN is Blank, but indicated it should be there
+            return true if item.ssn.blank? && item.ssn_data_quality == 1
+            # SSN is present but DQ indicates it shouldn't be
+            return true if item.ssn.present? && ![1, 2].include?(item.ssn_data_quality)
+            # SSN all zeros
```
Comment (awisema) [inline drivers/hmis_data_quality_tool/app/models/hmis_data_quality_tool/client.rb:336]: Interesting that some of the HUD specs have more complex SSN validity rules than others -- I assume this came from the workbook. Would it be better for this to match the Excel, or use `HUD.valid_social?` to match the HUD standards?

**Review by awisema — CHANGES_REQUESTED** [review]

File: db/warehouse/migrate/20221006193112_add_summary_fields_to_dqt_report.rb (hunk 1 above)
Comment (eanders) [reply db/warehouse/migrate/20221006193112_add_summary_fields_to_dqt_report.rb]: yes!

File: drivers/hmis_data_quality_tool/app/models/hmis_data_quality_tool/client.rb (hunk 2 above)
Comment (eanders) [reply drivers/hmis_data_quality_tool/app/models/hmis_data_quality_tool/client.rb:336]: So in this case, we're not checking for validity (though maybe we should.)  The idea of checking all zeros was that some systems use that to indicate blank if they can't handle actually saving blanks.

**Pull request merged**